Sprint_12

MiraiChan opened the pull request:
Done.

MiraiChan committed:
Class ImagesListService was added.

MiraiChan committed:
New file Photo Structures was added. Slight implementations in ImagesListService .

MiraiChan committed:
func fetchPhotosNextPage() is done according to the technical specification.

MiraiChan committed:
Replaced mock data with the actual data from the server for @IBAction didTapLikeButton.
Implemented like functionality.

MiraiChan committed:
IBOutlets and IBActions were added and linked to SingleImageViewController.
Some constant renaming were made in func rescaleAndCenterImageInScrollView().

MiraiChan committed:
In fun downloadImage()  was added the ability to view an enlarged image.
func showError() was created and added.

MiraiChan committed:
ProgressHUD was styled according to application specifics by executing code in AppDelegate.

MiraiChan committed:
New func cleanCookies() was added that clears cookies to log out from the user profile.

MiraiChan committed:
Alert was added to ProfileViewController before logout.
Second button "No" was added to Alert.
didTapButton was refactored.

MiraiChan committed:
Gradient is added to a cell. Constants changed to lower case.

DanilaG commented on ImageFeed/Services/ImagesListService.swift:
```diff
@@ -0,0 +1,140 @@
+        return Photo(
+            id: photoResult.id,
+            size: CGSize(width: Double(photoResult.width), height: Double(photoResult.height)),
+            createdAt: ISO8601DateFormatter().date(from: photoResult.createdAt ?? ""),
```
Давай сохраним date formatter в переменную, так как [согласно статье](https://sarunw.com/posts/how-expensive-is-dateformatter/) создовать форматтеры давольно таки затратно)

DanilaG replied:
`dateFormatter.date(from: photoResult.createdAt ?? "")` можно представить следующим образом:
`photoResult.createdAt.flatMap { dateFormatter.date(from: $0) }` что позволит не использовать пустую строку + упрощает понимание логики при чтении)

DanilaG commented on ImageFeed/Services/ImagesListService.swift:
```diff
@@ -0,0 +1,140 @@
+
+final class ImagesListService {
+    static let shared = ImagesListService()
+    static let didChangeNotification = Notification.Name(rawValue: "ImagesListServiceDidChange")
```
🔥 что с маленькой буквы называется константа)

DanilaG commented on ImageFeed/Services/ImagesListService.swift:
```diff
@@ -0,0 +1,140 @@
+    }
+    
+    func makeNextPageNumber() -> Int {
+        guard let lastLoadedPage else { return 1 }
```
Классно, что используешь фишки нового swift)

MiraiChan committed:
Constant for ISO8601DateFormatter() is created.
Files were organised in folders.
Some constant names in DateFormatter were changed so not get confused.
view.background was added as asked reviewer.

MiraiChan committed:
Info.plist was placed back outside the folder.
